airbyte-ci/connectors/pipelines/pipelines/models/contexts/pipeline_context.py

```diff
@@ -37,26 +39,65 @@ class PipelineContext:
     dockerd_service: Optional[Service]
     started_at: Optional[datetime]
     stopped_at: Optional[datetime]
+    _default_excluded_files: Optional[list[str]]
+    _faster_excluded_files: Optional[list[str]]
 
     secrets_to_mask: List[str]
 
     PRODUCTION = bool(os.environ.get("PRODUCTION", False))  # Set this to True to enable production mode (e.g. to send PR comments)
 
-    DEFAULT_EXCLUDED_FILES = (
-        [".git", "airbyte-ci/connectors/pipelines/*"]
-        + glob("**/build", recursive=True)
-        + glob("**/.venv", recursive=True)
-        + glob("**/secrets", recursive=True)
-        + glob("**/__pycache__", recursive=True)
-        + glob("**/*.egg-info", recursive=True)
-        + glob("**/.vscode", recursive=True)
-        + glob("**/.pytest_cache", recursive=True)
-        + glob("**/.eggs", recursive=True)
-        + glob("**/.mypy_cache", recursive=True)
-        + glob("**/.DS_Store", recursive=True)
-        + glob("**/airbyte_ci_logs", recursive=True)
-        + glob("**/.gradle", recursive=True)
-    )
+    def faster_get_default_excluded_files(self):
```

Comment:
nit: You could use a [ `@lru_cache`](https://docs.python.org/dev/library/functools.html#functools.lru_cache) decorator to memoize this function and not assign to ` self._faster_excluded_files`